sklearn/decomposition/ksvd.py

```diff
@@ -0,0 +1,343 @@
+"""K-SVD dictionary learning algorithm.
+"""
+
```

Comment:
Technically it is not hard to merge the functions completely and make possible every combination of initialization (SVD or random samples), sparse coding (lars, omp, ...) and dictionary updating (CD or K-SVD). Then the problem statement (L1 or L0) and parameters type (alpha or n_nonzero_coefs) depend on the sparse coding method.

However it may seem confusing for users, as some combinations do not correspond to well-known dictionary learning methods. We can still make aliases like `ksvd` that use canonical combinations.